driver/scsi_operation.c

```diff
@@ -605,6 +605,24 @@ WnbdPendOperation(_In_ PWNBD_EXTENSION DeviceExtension,
             FALSE);
         }
         break;
+    case SCSIOP_PERSISTENT_RESERVE_IN:
+    case SCSIOP_PERSISTENT_RESERVE_OUT:
+        ULONG DataTransferLength = SrbGetDataTransferLength(Srb);
+
+        if (DataTransferLength > WNBD_DEFAULT_MAX_TRANSFER_LENGTH)
+        {
+            Srb->SrbStatus = SRB_STATUS_ABORTED;
+            Status = STATUS_BUFFER_TOO_SMALL;
+            break;
+        }
+
+        // TODO: which size do we specify here? Is it the allocation
```

Comment:
TODO: keep data transfer length but ensure that the allocation length doesn't exceed it.